src/common/Eval.res

```diff
@@ -0,0 +1,65 @@
+type evalResult = result<string, string>
+
+@val external importMetaUrl: string = "import.meta.url"
```

Comment:
`import.meta.url` is currently evaluating to `undefined`. It looks like this is only supported in webpack 5 but from looking at the config this seems to be enabled? Not sure where to go from here...

Reply:
Ok, looks like it's esbuild transforming `import.meta` to an empty object. Setting `target: 'es2020'` in esbuild-loader fixes the issue. I'm not sure if this is appropriate though as I don't know what browser support the playground is targeted towards.

Reply:
@ryyppy can you help here at all?

Reply:
should we remove the esbuild loader setup?

Reply:
or why do you actually need that `import.meta.url` thing? I feel like the entry point for the worker needs to be located in the `/static` directory, so you can refer to it via `/static/evalworker.js` or something.. like the approach done in the original playground here: https://github.com/reasonml/reasonml.github.io/blob/source/website/static/js/evalWorker.js

Reply:
> or why do you actually need that `import.meta.url` thing?

This is in essence required only to be able to write the worker code in ReScript, or more specifically to be able to build the worker code with webpack5.

If we think the worker doesn't need to be written in ReScript then, as you say, the code can just be an un-transpiled JS file in /static.

I think it's useful to be write this worker in ReScript, being able to share the message types between app and worker is powerful. Could help with worker versioning too.

Reply:
Okay, so you want webpack to build the worker code for you.

Anyways, if setting `target: "es2020"` worked for you, does removing the esbuild plugin help getting a saner webpack behavior for webworkers? If so, we can undo these changes.

Reply:
> Anyways, if setting target: "es2020" worked for you, does removing the esbuild plugin help getting a saner webpack behavior for webworkers?

Both of these methods seem to work the same here. What do you mean by "saner webpack behavioue"?

Reply:
NextJS uses babel by default to transpile the source code. It might be that babel / webpack do some magic by transforming the `import.meta.url` to the correct constant? 

Anyways, I just added the code from the `with-webworker` template, removed the esbuild loader settings and it seems to load the webworker correctly:

![Uploading image.png…]()

Reply:
I can't see the screenshot, could share the diff? I reverted back to this commit and still having issues with worker code running in the window: https://github.com/rescript-association/rescript-lang.org/blob/8b1a3955baecabede5235492bc685b74233b95cc/next.config.js#L36-L61

Reply:
Here's the diff: https://github.com/rescript-association/rescript-lang.org/compare/test-webworker

Reply:
Ok, that's exactly what I've done 🤔 I'll push it to this branch so you can test

Reply:
I adapted my worker code that runs the PI worker example in ReScript:
https://github.com/rescript-association/rescript-lang.org/compare/test-webworker

Reply:
Going to resolve this thread as I think we've reached consensus on the original discussion ie. remove esbuild or keep it and set target: es2020.